indico/modules/events/registration/client/js/components/ConsentToPublishEditor.jsx

```diff
@@ -0,0 +1,69 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import registrationChangeConsent from 'indico-url:event_registration.api_registration_change_consent';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Icon} from 'semantic-ui-react';
+
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+import ConsentToPublishDropdown, {publishModePropType} from './ConsentToPublishDropdown';
+
+import './ConsentToPublishEditor.module.scss';
+
+export default function ConsentToPublishEditor({
+  locator,
+  publishToParticipants,
+  publishToPublic,
+  initialConsentToPublish,
+}) {
+  const [consentToPublish, setConsentToPublish] = useState(initialConsentToPublish);
+  const [loading, setLoading] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [failure, setFailure] = useState(false);
```

Comment:
this looks like a good usecase for `useDispatch` since you always change multiple fields at once depending on some action. or a much more simple alternative: a single useState for a 'status', ie you'd then do `setStatus(...)` with 'loading', 'success' or 'error'

Reply:
Thanks! I wanted to use an enum but I didn't find a good way to do it in JS - the strings for the states seem to be the best option